Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2765,6 +2766,14 @@ public void onActivityStarted(Activity activity) {
                 initSessionWithData(intentData, activity); // indicate  starting of session.
             }
             activityCnt_++;
+
+            // Activity stack for finding the structured path to this activity
+            if (activityStack_.size() > 1 && (activityStack_.get(activityStack_.size() - 2)).equals(activity.getClass().getSimpleName())) {
+                activityStack_.remove(activityStack_.get(activityStack_.size() - 2));
+                activityStack_.remove(activityStack_.get(activityStack_.size() - 1));
```

Comment:
Is there any risk that this gets called from two threads at once? Or is it only called on the main thread?